src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -387,19 +399,271 @@ export class PatientViewPageStore {
+                return Promise.resolve(
+                    this.molecularProfilesInStudy.result.filter(
+                        (profile: MolecularProfile) => {
+                            if (profile.genericAssayType) {
+                                return (
+                                    profile.genericAssayType ===
+                                    GenericAssayTypeConstants.MUTATIONAL_SIGNATURE
+                                );
+                            }
+                            return false;
+                        }
+                    )
+                );
+            },
+        },
+        []
+    );
+
+    readonly fetchAllMutationalSignatureData = remoteData(
+        {
+            await: () => [
+                this.samples,
+                this.mutationalSignatureMolecularProfiles,
+            ],
+            invoke: () => {
+                let sampleMolecularIdentifiers: SampleMolecularIdentifier[] = [];
+                const mutationalSignatureMolecularProfileIds = this.mutationalSignatureMolecularProfiles.result.map(
+                    profile => profile.molecularProfileId
+                );
+                if (mutationalSignatureMolecularProfileIds.length > 0) {
+                    _.forEach(
+                        mutationalSignatureMolecularProfileIds,
+                        mutationalSignatureMolecularProfileId => {
+                            const filters = _.map(
+                                this.samples.result,
+                                sample => {
+                                    return {
+                                        molecularProfileId: mutationalSignatureMolecularProfileId,
+                                        sampleId: sample.sampleId,
+                                    } as SampleMolecularIdentifier;
+                                }
+                            );
+                            sampleMolecularIdentifiers.push(...filters);
+                        }
+                    );
+                    return client.fetchGenericAssayDataInMultipleMolecularProfilesUsingPOST(
+                        {
+                            genericAssayDataMultipleStudyFilter: {
+                                sampleMolecularIdentifiers,
+                            } as GenericAssayDataMultipleStudyFilter,
+                        }
+                    );
+                }
+                return Promise.resolve([]);
+            },
+        },
+        []
+    );
+
+    readonly mutationalSignatureDataGroupByVersion = remoteData(
+        {
+            await: () => [
+                this.fetchAllMutationalSignatureData,
+                this.mutationData,
+                this.mutationalSignatureMetaGroupByStableId,
+            ],
+            invoke: () => {
+                const contributionData = this.fetchAllMutationalSignatureData.result.filter(
+                    data =>
+                        data.molecularProfileId.includes(
```

Comment:
be careful b/c includes is case sensitive.  do you want it to be case sensitive?